server/src/daemon/service.rs

```diff
@@ -24,11 +24,33 @@ use super::{
     collection::Collection, error::ServiceError, prompt::Prompt, session::Session, Result,
 };
 
+#[derive(Debug)]
+pub struct ServiceManager {
+    sessions: HashMap<OwnedObjectPath, Session>,
+}
+
+impl ServiceManager {
+    pub fn new() -> Self {
+        Self {
+            sessions: HashMap::new(),
+        }
+    }
+
+    pub fn sessions(&self) -> &HashMap<OwnedObjectPath, Session> {
+        &self.sessions
+    }
+
+    pub fn set_sessions(&mut self, path: OwnedObjectPath, session: Session) {
+        self.sessions.insert(path, session);
+    }
+}
+
```

Comment:
Does it make sense to put this inside a new file: `src/daemon/service_manager.rs`

Reply:
yeah, I agree :)